src/main/java/org/squiddev/plethora/integration/astralsorcery/MethodsAstralSorcery.java

```diff
@@ -0,0 +1,123 @@
+package org.squiddev.plethora.integration.astralsorcery;
+
+import dan200.computercraft.api.lua.LuaException;
+import hellfirepvp.astralsorcery.AstralSorcery;
+import hellfirepvp.astralsorcery.common.auxiliary.CelestialGatewaySystem;
+import hellfirepvp.astralsorcery.common.constellation.ConstellationRegistry;
+import hellfirepvp.astralsorcery.common.constellation.IConstellation;
+import hellfirepvp.astralsorcery.common.data.research.PlayerProgress;
+import hellfirepvp.astralsorcery.common.data.research.ResearchManager;
+import hellfirepvp.astralsorcery.common.data.research.ResearchProgression;
+import hellfirepvp.astralsorcery.common.data.world.data.GatewayCache;
+import hellfirepvp.astralsorcery.common.tile.TileCelestialGateway;
+import net.minecraft.entity.player.EntityPlayerMP;
+import net.minecraftforge.fml.relauncher.Side;
+import org.squiddev.plethora.api.meta.TypedMeta;
+import org.squiddev.plethora.api.method.IContext;
+import org.squiddev.plethora.api.method.LuaList;
+import org.squiddev.plethora.api.method.wrapper.FromSubtarget;
+import org.squiddev.plethora.api.method.wrapper.PlethoraMethod;
+import org.squiddev.plethora.api.module.IModuleContainer;
+import org.squiddev.plethora.gameplay.modules.PlethoraModules;
+import org.squiddev.plethora.integration.EntityIdentifier;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public final class MethodsAstralSorcery {
+	private MethodsAstralSorcery() {
+	}
+
+	//REFINE This method is potentially OP, as it exposes all available `GatewayNode`s.
+	// That said, players can already name a gateway, and they could add signs to the destination
+	// so that it shows in the preview (if it is working; haven't tested)
+	// If Hellfire complains, it's easy enough to delete...
+	@PlethoraMethod(
+		modId = AstralSorcery.MODID,
+		doc = "-- Get a list of all Celestial Gateways, grouped by dimension"
+	)
+	public static Map<String, ?> getGateways(@Nonnull IContext<TileCelestialGateway> context ) {
```

Comment:
If we're not using the context here, can we replace this with a `@FromTarget TileCelestialGateway`?